### PR TITLE
Support grad updates represented as IndexedSlices.

### PR DESCRIPTION
**Pull request by aaron276h**

### CODE REVIEW

File: horovod/_keras/__init__.py [hunk 1]
```diff
@@ -176,12 +186,14 @@ def allreduce_grads():
...
-                        lambda: self.grads,
-                    )
+                    if self._aggregation_frequency > 1:
+                        return tf.cond(
+                            tf.equal(self.counter, self._aggregation_frequency),
+                            allreduce_grads,
+                            lambda: self.grads,
+                        )
+                    else:
+                        return allreduce_grads()
```
Comment (bfrieds) [inline horovod/_keras/__init__.py]: what is the motivation behind this change?

File: horovod/_keras/__init__.py [hunk 2]
```diff
@@ -155,8 +158,15 @@ def allreduce_grads():
...
-                        return [tf.divide(g, self._aggregation_frequency) for g in averaged_gradients]
+                    if self._aggregation_frequency > 1:
+                        with tf.control_dependencies([reset_op]):
+                            return [tf.divide(g, self._aggregation_frequency) for g in averaged_gradients]
+                    else:
+                        # When grad updates are represented in `IndexedSlices`, we can not divide
+                        # them by int. Currently _aggregation_frequency > 1 is not supported
+                        # `IndexedSlices`.
+                        with tf.control_dependencies([reset_op]):
+                            return [tf.identity(g) for g in averaged_gradients]
```
[the remark this answers is not in the document]
Comment (aaron276h) [reply horovod/_keras/__init__.py]: Yeah this is definitely better, I updated accordingly.

File: horovod/_keras/__init__.py (hunk 1 above)
Comment (aaron276h) [reply horovod/_keras/__init__.py]: For some reason when running wtth `CapsuleNet` it did not consider `allreduce_grads` and `lambda: self.grads` to be equivalent graphs, and would error out.
